### PR TITLE
.github: Add PR and Issue templating

### PR DESCRIPTION
**Pull request by craciunoiuc**
The template is imported and modified from the main unikraft repository.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+### Prerequisite checklist
+
+<!--
+Please mark items appropriately:
+-->
+
+ - [ ] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
+ - [ ] Tested your changes against relevant architectures and platforms;
+ - [ ] Ran `go fmt` on your commit series before opening this PR;
```
Comment (nderjung) [inline .github/PULL_REQUEST_TEMPLATE.md]: `make fmt`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ - [ ] Updated relevant documentation.
+
+### Base target
+
+ - Architecture(s): [e.g. `x86_64`, `arm`, `arm64`, or N/A]
```
Comment (nderjung) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can delete

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ - Operating System(s): [e.g. `windows`, `linux`, `macOS`, or N/A]
```
Comment (nderjung) [inline .github/PULL_REQUEST_TEMPLATE.md]: Not yet relevant, can remove

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        Thanks for taking the time to fill out this bug report!
+
+        Please only use this form to submit problems with building Kraftkit or it's interaction with Unikraft unikernels.  This includes unexpected or undefined behavior, stacktraces, critical messages, and things just breaking or exploding.
+        
+        **Ensure you have tested against the latest changes on `staging` branch before submitting this form.**
+
+        For questions, concerns or help with running Kraftkit, please check out our additional resources:
+
+        * [Unikraft Documentation](https://unikraft.org/docs/)
+        * [Unikraft Discussions Forum](https://github.com/unikraft/kraftkit/discussions)
```
Comment (nderjung) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: We don't have a discussions forum for kraftkit, so we can switch it back to the unikraft core repo for discussions

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  - type: dropdown
+    id: platforms
+    attributes:
+      label: Which operating system were you using or does this bug affect?
+      description: If this bug is operating system agnostic or you are unsure then please do not select an platform listed below.
+      multiple: true
+      options:
+        - linux
+        - windows
+        - macOS
```
Comment (nderjung) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: let's be more specific here and only add the platform's we're currently supporting:
```
- linux/debian
- linux/fedora
- linux/alpine
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+blank_issues_enabled: false
+
+contact_links:
+  - name: 💡 Discuss an idea
+    url: https://github.com/unikraft/kraftkit/discussions/new
```
Comment (nderjung) [inline .github/ISSUE_TEMPLATE/config.yml]: Switch to unikraft core repo discussions forum

File: .github/ISSUE_TEMPLATE/project_backlog.yml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+description: Create an issue for new work to be done which has been discussed.
+labels: [kind/enhancement]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Please only use this form to submit new, well-defined work to the Kraftkit project's backlog.  Issues under this category have been discussed with the Kraftkit team.
+
+        If you're intending to request a feature, please [open a discussion using the "Ideas" category](https://github.com/unikraft/kraftkit/discussions/new) instead so we can understand your workflow first.  If you are unsure about a feature, please [check out our documentation](https://docs.unikraft.org/) which covers existing features and possibilities with Unikraft or filter existing issues labeled under [`kind/enhancement`](https://github.com/unikraft/kraftkit/labels/kind/enhancement) to prevent double-posting.
+  - type: textarea
```
Comment (nderjung) [inline .github/ISSUE_TEMPLATE/project_backlog.yml]: Same here re:discussions forum

File: .github/ISSUE_TEMPLATE/project_backlog.yml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  - type: dropdown
+    id: related-platforms
+    attributes:
+      label: Related platforms
+      description: Please indicate whether the feature request is related to a specific operating system.  If this feature is operating system agnostic or you are unsure then please do not select an operating system listed below.
+      multiple: false
+      options:
+        - linux
+        - windows
+        - macOS
```
Comment (nderjung) [inline .github/ISSUE_TEMPLATE/project_backlog.yml]: same again with os disbutions

Comment (nderjung) [reply .github/ISSUE_TEMPLATE/project_backlog.yml]: maybe also we can add "other"

**Review by nderjung — CHANGES_REQUESTED** [review]
All good, just address the few inline issues + the codacy issue with the whitespace before the `-` in markdown lists

**Comment by craciunoiuc** [comment]
I suggest we just ignore the files inside the templating, it seems I can't please Codacy here

<https://docs.codacy.com/repositories-configure/ignoring-files/>

**Comment by nderjung** [comment]
> I suggest we just ignore the files inside the templating, it seems I can't please Codacy here
> 
> https://docs.codacy.com/repositories-configure/ignoring-files/

I agree. This can be sorted in a separate PR.

**Review by nderjung — APPROVED** [review]
Looks good, thanks!

**Pull request merged**